### PR TITLE
[Upbit] add getAccountInfo

### PR DESCRIPTION
**Pull request by water0126**

### CODE REVIEW

File: xchange-upbit/pom.xml [hunk 1]
```diff
@@ -28,7 +28,10 @@
             <artifactId>xchange-core</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.auth0</groupId>
+            <artifactId>java-jwt</artifactId>
```
Comment (timmolter) [inline xchange-upbit/pom.xml]: can you define this in the parent pom instead?

Comment (water0126) [reply xchange-upbit/pom.xml]: yes, I changed location. and java-jwt version upgraded.

File: pom.xml [hunk 2]
```diff
@@ -286,13 +287,17 @@
...
 			</dependency>
-
-
 		</dependencies>
 	</dependencyManagement>
 
 	<!-- All children will have these dependencies -->
 	<dependencies>
+		<!-- jwt for auth -->
+		<dependency>
```
Comment (timmolter) [inline pom.xml]: this should go under dependency management since not all modules require it.

Comment (water0126) [reply pom.xml]: sorry, modified

File: xchange-upbit/src/main/java/org/knowm/xchange/upbit/service/UpbitJWTDigest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        Map<String, String> map = mapper.readValue(restInvocation.getRequestBody(), Map.class);
+        it = map.keySet().iterator();
+        while (it.hasNext()) {
+          String key = (String) it.next();
+          String value = map.get(key);
+          queryString += "&" + key + "=" + value;
+        }
+        queryString = queryString.substring(1);
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (walec51) [inline xchange-upbit/src/main/java/org/knowm/xchange/upbit/service/UpbitJWTDigest.java]: wrap the IOException in a IllegalStateException and throw it out 
there is no sense in sending a request with a broken digset

File: xchange-upbit/src/main/java/org/knowm/xchange/upbit/service/UpbitJWTDigest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    Iterator it;
+    if (restInvocation.getParamsMap().get(QueryParam.class) != null
+        && !restInvocation.getParamsMap().get(QueryParam.class).isEmpty()) {
+      queryString = String.valueOf(restInvocation.getParamsMap().get(QueryParam.class));
+    } else if (restInvocation.getRequestBody() != null
+        && !restInvocation.getRequestBody().isEmpty()) {
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        Map<String, String> map = mapper.readValue(restInvocation.getRequestBody(), Map.class);
+        it = map.keySet().iterator();
```
Comment (walec51) [inline xchange-upbit/src/main/java/org/knowm/xchange/upbit/service/UpbitJWTDigest.java]: why is it declared earlier if it is used only here?

File: xchange-upbit/src/main/java/org/knowm/xchange/upbit/dto/account/UpbitBalance.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package org.knowm.xchange.upbit.dto.account;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.math.BigDecimal;
+
+public class UpbitBalance {
+  private final String currency;
+  private final BigDecimal balance;
+  private final BigDecimal locked;
+  private final BigDecimal avg_krw_buy_price;
```
[the remark this answers is not in the document]
Comment (timmolter) [reply xchange-upbit/src/main/java/org/knowm/xchange/upbit/dto/account/UpbitBalance.java]: agreed

**Pull request merged**